File: 0009-palindrome-number/0009-palindrome-number.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
         string num_str = to_string(x);
        
        
        int left = 0;
        int right = num_str.size() - 1;
        
       
        
        while(left < right){
            if(num_str[left] != num_str[right]){
                return false;
            }
                
            left += 1;
            right -= 1;
        }
            
        
        return true;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        
        num_str = str(x)
        
        
        left, right = 0, len(num_str) - 1
        
       
        while left < right:
            if num_str[left] != num_str[right]:
                return False
            left += 1
            right -= 1
        
        return True
                    
        